### ISSUE TITLE
Is it possible to create a form with recursive field?

### ISSUE THREAD

**Issue by mattp0123** [fabian-hiller/modular-forms#251]
I have a form like this:
```typescript
type Condition =
  | {
      type: 'composite'
      operator: 'and' | 'or'
      conditions: Condition[]
    }
  | OtherCondition

type Form = {
  condition: Condition
}
```

I got this error when i trying to create the form:

![image](https://github.com/user-attachments/assets/2b5c9d83-93f5-4771-a8ec-aa6b96ddf655)

**Comment by fabian-hiller** [comment]
Hmm... the recursive type probably breaks the types of the library. 😐

**Comment by mattp0123** [comment]
@fabian-hiller maybe this could help: https://dev.to/scooperdev/supporting-circularly-referenced-mapped-types-in-typescript-4825

**Comment by fabian-hiller** [comment]
Yes, this might work, but it limits the depth. You can do this yourself by explicitly specifying a depth of only 10 instead of infinity in your schema.
